parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -121,12 +131,23 @@ private ParquetMetadataConverter(boolean useSignedStringMinMax) {
       cachedEncodingSets = new ConcurrentHashMap<Set<org.apache.parquet.column.Encoding>, Set<org.apache.parquet.column.Encoding>>();
 
   public FileMetaData toParquetMetadata(int currentVersion, ParquetMetadata parquetMetadata) {
+    try {
+      return toParquetMetadata(currentVersion, parquetMetadata, (PositionOutputStream) null, (ParquetFileEncryptor) null);
+    } catch (IOException e) {
+      // TODO change the method signature in maven build
```

Comment:
This IOException is thrown only during encryption (with a valid encryptor), so should never happen here since the method is called with a null encryptor. But, if runtime exceptions are acceptable in this project, throwing one is indeed cleaner than returning a null. Will change accordingly.

Reply:
Very descriptive exception naming right? :)

Reply:
Its perfect.. Makes me wonder how come Java doesn't have it as a part of the standard toolkit.